stores/amazon.py

```diff
@@ -462,9 +480,92 @@ def check_stock(self, asin, reserve_min, reserve_max, retry=0):
 
         timeout = self.get_timeout()
         while True:
+            # Sanity check to see if we have any offers
+            try:
+                offers_exist = WebDriverWait(
+                    self.driver, timeout=DEFAULT_MAX_TIMEOUT
+                ).until(
+                    lambda d: d.find_element_by_xpath(
+                        "//div[@id='aod-container'] | "
+                        "//div[@id='olpOfferList'] | "
+                        "//span[@data-action='show-all-offers-display'] | "
+                        "//div[@id='backInStock' or @id='outOfStock'] |"
+                        "//input[@name='submit.add-to-cart' and not(//span[@data-action='show-all-offers-display'])]"
+                    )
+                )
+                offer_count = []
+                offer_id = offers_exist.get_attribute("id")
+                if offer_id == "outOfStock" or offer_id == "backInStock":
+                    # No dice... Early out and move on
+                    log.info("Item is currently unavailable.  Moving on...")
+                    return False
+
+                if offer_id == "olpOfferList":
+                    # Offers Page ... count the 'a-row' classes to know how many offers we 'see'
+                    offer_count = self.driver.find_elements_by_xpath(
+                        "//div[@id='olpOfferList']//div[contains(@class, 'olpOffer')]"
+                    )
+                elif offer_id == "aod-container":
+                    # Offer Flyout or Ajax call ... count the 'aod-offer' divs that we 'see'
+                    offer_count = self.driver.find_elements_by_xpath(
+                        "//div[@id='aod-pinned-offer' or @id='aod-offer']"
+                    )
+                elif (
+                    offers_exist.get_attribute("data-action")
+                    == "show-all-offers-display"
+                ):
+                    # No offers to parse... look for a link to the offers
+                    log.info("Attempting to click the open offers link...")
+                    self.driver.find_element_by_xpath(
+                        "//span[@data-action='show-all-offers-display']//a"
+                    ).click()
```

Comment:
I am getting this error below
```
selenium.common.exceptions.ElementClickInterceptedException: Message: element click intercepted: Element <a id="buybox-see-all-buying-choices-announce" href="/gp/offer-listing/B08P2H5LW2/ref=dp_olp_unknown_mbc" title="See All Buying Options" class="a-button-text" role="button">...</a> is not clickable at point (1111, 306). Other element would receive the click: <form method="post" action="/gp/add-to-cart/html/ref=aod_dpdsk_new_1" class="aod-atc-form-header-desktop a-spacing-none">...</form>
```
Instead using this seems to be working for me
```
element = self.driver.find_element_by_xpath("//span[@data-action='show-all-offers-display']//a")
                    self.driver.execute_script("arguments[0].click();", element)
```
ref: https://stackoverflow.com/questions/48665001/can-not-click-on-a-element-elementclickinterceptedexception-in-splinter-selen

Reply:
can confirm this is also what worked for me, even making the window take the entire screen the fly out obscures the button behind it

Reply:
Are you guys getting to the PDP with the flyout already open?  We can use javascript to click the button, but if the button click is just to open the flyout that's already open, there's no actual need to click it.  We just need to prevent the exception in that case since it's already opened.  However, no one has reported what the browser looks like or what the HTML is in those scenarios.  The Devs are slowing tracking this down, but if you guys have examples of what is actually happening on the page, that would help more than finding alternate ways to click an obscured button.  The thing that's obscuring the button is the reason we're clicking the button!

Any additional help is appreciated.

Reply:
yep, the redirect from the original URL puts you in the PDP with the flyout already open

Reply:
tried with edge/firefox/brave (logged in), no redirect (old offer-listing page)
tried incognito firefox/brave without logging in, they all redirected me to the new page with no automated fly-out
tried incognito edge without logging in, no redirect (old offer-listing page)

tried with the chrome window spawned by the script, it did redirect + automatic flyout. tried incognito and no redirect (still old offer-listing page)


I can't see any pattern, sometimes it redirects and sometimes it does a fly-out automatically